### PR TITLE
adapter: Remove old migrations

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation
This PR refactors existing code.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by jkosh44** [comment]
See: https://github.com/MaterializeInc/materialize/issues/17466

File: src/adapter/src/catalog/storage.rs
Comment (mjibson) [inline src/adapter/src/catalog/storage.rs:375]: Can RoleAttribute be made into a non-Option now?

Comment (jkosh44) [reply src/adapter/src/catalog/storage.rs:375]: Done

**Comment by jkosh44** [comment]
It looks like the upgrade tests are testing versions earlier than v45, so we'll have to wait until the next release to merge.

File: src/environmentd/src/lib.rs
Comment (teskje) [inline src/environmentd/src/lib.rs:355]: I forgot about this, thanks!

**Review by teskje — APPROVED** [review]
LGTM!

**Comment by jkosh44** [comment]
It looks like the upgrade tests are testing versions earlier than v45, so we'll have to wait until the next release to merge.

> It looks like the upgrade tests are testing versions earlier than v45, so we'll have to wait until the next release to merge.

Turns out we have to wait for one more release: https://materializeinc.slack.com/archives/C01LKF361MZ/p1678816536147819

**Pull request merged**
